relay-server/src/actors/global_config.rs

```diff
@@ -254,23 +258,43 @@ impl Service for GlobalConfigService {
             let mut shutdown_handle = Controller::shutdown_handle();
 
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match self.config.relay_mode() {
+                RelayMode::Managed => {
+                    if self.config.has_credentials() {
+                        relay_log::info!("serving global configs fetched from upstream");
+                        self.request_global_config();
+                    } else {
+                        relay_log::info!(
+                            "serving default global configs due to lacking credentials"
+                        );
```

Comment:
doing nothing means serving default config, because we initiate the watch with default in new();